src/Guesses.jsx

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from "react";
+import "./Guesses.css";
+
+//basing this off data from guessArray coming in to be ['guess string', '25' - 25 being the percentage correct] - Hopefully that works
+
+export default function Guesses({ guessArray }) {
+  const [boxes, setBoxes] = useState(Array(8).fill(""));
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    if (guessArray && guessArray.length === 2 && guessArray[0] !== "") {
+      const [guess, numberStr] = guessArray;
+      const number = parseInt(numberStr);
+      console.log("number", number);
+
+      setBoxes((prevBoxes) => {
+        const newBoxes = [...prevBoxes];
+        newBoxes[currentIndex] = (
+          <div className={`box ${getColorClass(number)}`}>{guess}</div>
+        );
+        return newBoxes;
+      });
+
+      setCurrentIndex((prevIndex) => (prevIndex + 1) % 8);
+    }
+  }, [guessArray]);
```

Comment:
I tested your code around and see that I pass in an guessArray with the syntax `["guess", accuracy number]`. Every time someone makes a new guess you pass in that new guess as a different `guessArray` prop and the new guess will be inserted into the right place by moving index forward.

While this works, I think it is better if we do not do this as our component is keeping track of it's own state.
1. If we need this list of guesses anywhere else , we cannot pass it upwards to it's parent component `<Game />` (where the game logic lives) due to how react works. Any operations that require this list (ex: saving what a player guessed for a certain game) would need to work around not having access to this data easily in any sibling components (as data cannot move horizontally). 
2. If we want to keep this way, another solution would be to keep track of the current guesses in the parent component `<Game />` and this component simultaneously. I do not think it be a good idea because we should avoid keeping track of our data model in two different places and having to keep them in sync, as things can get out of hand.

The better solution I think would be passing in an array of `guessArray`s. For example:
```js
// one guess
guessArray = [
  ["Dog", 12],
]
// two guesses
guessArray = [
  ["Dog", 12],
  ["Cat", 20],
]
// three guesses
guessArray = [
  ["Dog", 12],
  ["Cat", 20],
  ["Lion", 50],
]
// ... etc till you hit 8 guesses
```

The reason I think this is better is because we can have this guess history state living in our parent component `<Game />` and it will pass this into our `<Guesses />` component. Guesses will not need to keep track of what previous guesses were like they are currently are. We will tell our component every time, what the entire history is.

Reply:
I'll see what I can do